src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,176 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    private class Node<E> {
+        private E value;
+        private Node<E> next;
+        private Node<E> prev;
+
+        public Node(Node<E> prev, E value, Node<E> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (head == null) {
+            head = node;
+            tail = node;
+        } else {
+            tail.next = node;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
```

Comment:
in case if `index == size` let's use `add(T value)` method here before index checking

Reply:
can't do this. received: testAddByIndexToTheTop(core.basesyntax.MyLinkedListTest): Expected test to throw an instance of java.lang.IndexOutOfBoundsException
  testAddByIndexToTheBottom(core.basesyntax.MyLinkedListTest): Expected test to throw an instance of java.lang.IndexOutOfBoundsException